packages/contracts/test/asset_proxy_dispatcher/dispatcher.ts

```diff
@@ -0,0 +1,297 @@
+    let erc20Proxy: ERC20ProxyContract;
+    let erc721Proxy: ERC721ProxyContract;
+    const INITIAL_BALANCE = new BigNumber(10000);
+
+    before(async () => {
+        // Setup accounts & addresses
+        const accounts = await web3Wrapper.getAvailableAddressesAsync();
+        owner = tokenOwner = accounts[0];
+        notOwner = notAuthorized = accounts[1];
+        exchangeAddress = accounts[2];
+        makerAddress = accounts[3];
+        takerAddress = accounts[4];
+        // Deploy Asset Proxy Dispatcher
+        const assetProxyDispatcherInstance = await deployer.deployAsync(ContractName.AssetProxyDispatcher);
+        assetProxyDispatcher = new AssetProxyDispatcherContract(
+            assetProxyDispatcherInstance.abi,
+            assetProxyDispatcherInstance.address,
+            provider,
+        );
+        await assetProxyDispatcher.addAuthorizedAddress.sendTransactionAsync(exchangeAddress, {
+            from: owner,
+        });
+        // Deploy ERC20 Proxy
+        const erc20ProxyInstance = await deployer.deployAsync(ContractName.ERC20Proxy);
+        erc20Proxy = new ERC20ProxyContract(erc20ProxyInstance.abi, erc20ProxyInstance.address, provider);
+        await erc20Proxy.addAuthorizedAddress.sendTransactionAsync(assetProxyDispatcher.address, {
+            from: owner,
+        });
+        // Deploy ERC721 Proxy
+        const erc721ProxyInstance = await deployer.deployAsync(ContractName.ERC721Proxy);
+        erc721Proxy = new ERC721ProxyContract(erc721ProxyInstance.abi, erc721ProxyInstance.address, provider);
+        await erc721Proxy.addAuthorizedAddress.sendTransactionAsync(assetProxyDispatcher.address, {
+            from: owner,
+        });
+        // Deploy zrx and set initial balances
+        const zrxInstance = await deployer.deployAsync(ContractName.DummyToken, constants.DUMMY_TOKEN_ARGS);
+        zrx = new DummyTokenContract(zrxInstance.abi, zrxInstance.address, provider);
+        await zrx.setBalance.sendTransactionAsync(makerAddress, INITIAL_BALANCE, { from: tokenOwner });
+        await zrx.setBalance.sendTransactionAsync(takerAddress, INITIAL_BALANCE, { from: tokenOwner });
+        dmyBalances = new Balances([zrx], [makerAddress, takerAddress]);
+        await zrx.approve.sendTransactionAsync(erc20Proxy.address, INITIAL_BALANCE, {
+            from: takerAddress,
+        });
+        await zrx.approve.sendTransactionAsync(erc20Proxy.address, INITIAL_BALANCE, {
+            from: makerAddress,
+        });
+    });
+    beforeEach(async () => {
+        await blockchainLifecycle.startAsync();
+    });
+    afterEach(async () => {
+        await blockchainLifecycle.revertAsync();
+    });
+    describe('addAssetProxy', () => {
+        it('should record proxy upon registration', async () => {
+            await assetProxyDispatcher.addAssetProxy.sendTransactionAsync(
+                AssetProxyId.ERC20,
+                erc20Proxy.address,
+                ZeroEx.NULL_ADDRESS,
```

Comment:
Set this to a descriptive variable name so next dev doesn't need to guess what this param refers to:
```
const prevProxyAddress = ZeroEx.NULL_ADDRESS;
```